api-reference/docstubs/api/calendar-post-calendarpermissions.md

```diff
@@ -0,0 +1,118 @@
+---
+title: "Add calendarPermissions"
+description: "Add calendarPermissions by posting to the calendarPermissions collection."
+author: ""
+localization_priority: Normal
+ms.prod: ""
+doc_type: apiPageType
+---
+
+# Add calendarPermissions
+
+Namespace: microsoft.graph
+
+Add calendarPermissions by posting to the calendarPermissions collection.
+
+## Permissions
+One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/concepts/permissions-reference.md).
+
+|Permission type|Permissions (from most to least privileged)|
+|:---|:---|
+|Delegated (work or school account)|**TODO: Determine scopes **|
+|Delegated (personal Microsoft account)|Not supported.|
+|Application|**TODO: Determine AppOnly scopes **|
+
+## HTTP request
+<!-- {
+  "blockType": "ignored"
+}
+-->
+``` http
+POST /me/messages/{messageId}/event/calendar/calendarPermissions/$ref
+```
+
+## Request headers
+|Name|Description|
+|:---|:---|
+|Authorization|Bearer {token}.Required|
+|Content-Type|application/json.Required|
+
+## Request body
+In the request body, supply a JSON representation for the [calendarPermission](../resources/calendarpermission.md) object.
+
+The following table shows the properties that are required when you create the [calendarPermission](../resources/calendarpermission.md).
+
+|Property|Type|Description|
+|:---|:---|:---|
+|id|String| Inherited from [entity](../resources/entity.md)|
+|emailAddress|[emailAddress](../resources/emailaddress.md)||
+|isRemovable|Boolean||
+|isInsideOrganization|Boolean||
+|role|Enumeration| Possible values are: `none`, `freeBusyRead`, `limitedRead`, `read`, `write`, `delegateWithoutPrivateEventAccess`, `delegateWithPrivateEventAccess`, `custom`.|
```

Comment:
I can't tell whether we should say Enumeration here, or the enumeration type name (without a link because we don't have enum topics).